app/ui/browser/views/browser.jsx

```diff
@@ -163,9 +163,7 @@ BrowserWindow.propTypes = {
 };
 
 const mapStateToProps = (state) => ({
-  // @TODO restructure the state so this list is automatically created as new todos
-  // are added instead of generated on each update pass
-  pageIds: state.browserWindow.pages.toArray().map(page => page.id),
+  pageIds: state.browserWindow.pageIds,
```

Comment:
Adding state to avoid a single `map` seems like the wrong trade-off to me.  Can you explain why this is better than the existing code, and better than an `OrderedMap` from `pageId` to `page`?